src/skmatter/utils/_pcovr_utils.py

```diff
@@ -186,7 +186,7 @@ def pcovr_covariance(
         C_Y = C_Y.reshape((C.shape[0], -1))
         C_Y = np.real(C_Y)
 
-        C += (1 - mixing) * C_Y @ C_Y.T
+        C += (1 - mixing) * np.array(C_Y @ C_Y.T, dtype=np.float64)
```

Comment:
Why is this necessary? It seems weird to suddenly enforce here float64 to me

Reply:
Artifact again!

Reply:
Without this we get  `numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'add' output from dtype('O') to dtype('float64') with casting rule 'same_kind'`

Reply:
I'll check this another time in more detail